lib/srv/app/server.go

```diff
@@ -1139,6 +1144,19 @@ func (s *Server) getProxyPort() string {
 	return port
 }
 
+// sessionStartTime fetches the session start time based on the the certificate
+// valid date.
+func (s *Server) sessionStartTime(ctx context.Context) time.Time {
+	var startTime time.Time
+	if userCert, err := authz.UserCertificateFromContext(ctx); err == nil {
+		startTime = userCert.NotBefore
+	} else {
+		s.log.Warn("Unable to retrieve session start time from certificate.")
+	}
+
+	return startTime
```

Comment:
Maybe: ditch the startTime var and return directly?

```suggestion
	if userCert, err := authz.UserCertificateFromContext(ctx); err == nil {
		return userCert.NotBefore
	}

	s.log.Warn("Unable to retrieve session start time from certificate.")
	return time.Time{}
```